### DISCUSSION TITLE
Best practices for managing server crashes

### DISCUSSION THREAD

**Discussion by jviotti** [uNetworking/uWebSockets#1628] [Q&A]
What are the best practices for deploying a uWebSockets application to GNU/Linux and handling the case where things go wrong (i.e. segfaults)? Are people using uWebSockets in production integrating with something like Crashpad + Sentry, or doing something else?

**Comment by uNetworkingAB** [comment]
That sounds like an oxymoron to me. Your server shouldn't crash, and if it does, you need to fix it. Use AddressSanitizer until you no longer segfault.

**Reply by jviotti** [reply]
Sure, but you often find these edge cases on production, and without any sort of crash reporting, then you would not know why they crashed on the first place?

**Reply by uNetworkingAB** [reply]
There is no built-in integration to any such monitor / logger. But you can do that on your own however you like it.
